enhancements/network/primary-user-defined-networks-for-virtualization-workloads.md

```diff
@@ -0,0 +1,638 @@
+fetch resources from outside networks (internet).
+- As a VM owner migrating my VMs to OCP+V, I do not want to change my
+application to support multiple NICs.
+- As a VM owner, I want to expose my selected applications over the network to
+users outside the cluster.
+- As an admin, I'm limited by public cloud networking restrictions and I rely
+on their LoadBalancer to route traffic to my applications.
+
+### Goals
+
+- The IP addresses on the VM must be the same before / after live-migration and
+VM restart / shutdown
+- Live-migration without breaking the established TCP connections
+- Provide a configurable way for the user to define the IP addresses on a VM's
+interface.
+- Native IPv6 integration
+- Integration with service meshes, and OpenShift observability solutions
+
+### Non-Goals
+
+TODO
+
+## Proposal
+
+To compartmentalize the solution, the proposal will be split in three different
+topics:
+- [Extending networking from the pod interface to the VM](#extending-networking-from-the-pod-interface-to-the-VM)
+- [Persisting VM IP addresses during the migration](#persisting-vm-ip-addresses-during-the-migration)
+- [Allow user to configure the VM's interface desired IP address](#allow-user-to-configure-the-vms-interface-desired-ip-address)
+
+Before that, let's ensure the proper context is in place.
+
+### Basic UDN context
+
+As indicated in the
+[user defined network segmentation proposal](https://github.com/openshift/enhancements/blob/master/enhancements/network/user-defined-network-segmentation.md#proposal),
+the pods featuring a primary user defined network will feature two interfaces:
+- an attachment to the cluster default network
+- an attachment to the primary user defined network
+
+Later on, in the
+[services section](https://github.com/openshift/enhancements/blob/master/enhancements/network/user-defined-network-segmentation.md#services-1),
+it is explained that all traffic will be send over the primary UDN interface.
+
+### Basic virtualization context
+
+In KubeVirt / OpenShift Virtualization, the VM runs inside a pod (named
+virt-launcher). Thus, given OVN-Kubernetes configures the pod interfaces (and
+is responsible for configuring networking up to the pod interface), we still
+need to extend connectivity from the pod interface into the VM.
+
+During live-migration - once it is is scheduled - and the destination
+node is chosen, a new pod is scheduled in the target node (let's call this pod
+the *destination* pod). Once this pod is ready, the *source* pod transfers the
+state of the live VM to the *destination* pod via a connection proxied by the
+virt-handlers (the KubeVirt agents running on each node) to the *destination*
+pod.
+
+### Extending networking from the pod interface to the VM
+
+KubeVirt uses a concept called bind mechanisms to extend networking to the VM;
+we plan on using the
+[passt](https://passt.top/passt/about/#passt-plug-a-simple-socket-transport)
```

Comment:
Following the recent talks about splitting the efforts to:
4.18: using tap/macvtap
4.19: using passt
Can you please update the plan?